doc/dev_guide/internals/box_protocol.rst

```diff
@@ -1281,16 +1282,20 @@ each call to conn:new_stream() assigns a new number, starting with 1.
 ..  _box_protocol-stream_transactions:
 
 The client makes stream transactions by sending, in order:
-IPROTO_BEGIN, the transaction data-change and query requests,
-IPROTO_COMMIT or IPROTO_ROLLBACK.
-Each request must contain the same IPROTO_STREAM_ID value.
-With streaming there is no need to add
-:ref:`IPROTO_FLAGS <box_protocol-flags>` and IPROTO_FLAG_COMMIT
-in the header of the last request of a transaction.
-Rollback will be automatic if disconnect occurs before commit is possible.
+
+1. IPROTO_BEGIN with an optional transaction timeout in the IPROTO_TIMEOUT field of the request body
```

Comment:
It's unclear what the type of `IPROTO_TIMEOUT` is.

Overall, I don't like how IPROTO transactions are documented. We have a separate entry for each IPROTO command (`IPROTO_SELECT`, `IPROTO_REPLACE`, etc) with the description of header and data fields, but not for `IPROTO_BEGIN`, `IPROTO_COMMIT`, `IPROTO_ROLLBACK` - those commands are clustered together in one section, but they are no worse than other commands so why would we want to treat them in a different way? If I developed a connector to Tarantool, I'd prefer all IPROTO commands and their arguments to be listed explicitly.

[![image](https://user-images.githubusercontent.com/1114511/172121309-05ffc99f-b557-42a9-ad14-9ec59b901798.png)](https://www.tarantool.io/en/doc/latest/dev_guide/internals/box_protocol/)

Also, we already have entries for `IPROTO_COMMIT` and `IPROTO_ROLLBACK`, but those are actually `IPROTO_RAFT_COMMIT` and `IPROTO_RAFT_ROLLBACK` - see the picture above. This is confusing. Better rename them there.

Reply:
> Overall, I don't like how IPROTO transactions are documented
Yeah, I've noticed this inconsistency. Filed a new issue (#2922), will do in this PR together with proper description of IPROTO_TIMEOUT.

> Also, we already have entries for IPROTO_COMMIT and IPROTO_ROLLBACK, but those are actually IPROTO_RAFT_COMMIT and IPROTO_RAFT_ROLLBACK
Thanks, fixed.